src/test/java/com/commercetools/sync/products/helpers/VariantReferenceResolverTest.java

```diff
@@ -317,8 +314,8 @@ void resolveReferences_WithMixedReferences_ShouldResolveReferenceAttributes() {
         resolvedCustomerReferenceAttribute.getValue();
     assertThat(resolvedCustomerReferenceAttributeValue).isNotNull();
 
-    assertThat(resolvedCustomerReferenceAttributeValue.get(REFERENCE_ID_FIELD))
-        .isEqualTo(customerReference.get(REFERENCE_ID_FIELD));
+    assertThat(resolvedCustomerReferenceAttributeValue.get(REFERENCE_ID_FIELD).asText())
```

Comment:
could you please add state reference in this test case?

Reply:
https://github.com/commercetools/commercetools-sync-java/pull/715/commits/b3b8f4d7148341377dd348f08719155d7d63ef04

Reply:
> Looks good, thanks for the extended coverage, I added some small comments.
> Additionally, it would be nice if you update product sync usage docs and double-check java docs.

https://github.com/commercetools/commercetools-sync-java/pull/715/commits/1bf86293652c1c56fd19effb4ebcda04fc1c11f4